core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutors.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.Flags;
+
+/**
+ * Provides a builder that is useful for creating an BlockingTaskExecutor.
+ *
+ * @see CommonPools#blockingTaskExecutor()
+ */
+public final class BlockingTaskExecutors {
```

Comment:
I actually meant this:

```java
public interface BlockingTaskExecutor extends ScheduledExecutorService {
    static BlockingTaskExecutor builder() { ... }
}
```

Reply:
I am not sure I understood fully what you intended to say.

It would be better to look at the code and get more ideas from it :)

Reply:
I guess what @trustin means is providing `BlockingTaskExecutor` interface and its builder:
```java
public interface BlockingTaskExecutor extends ScheduledExecutorService {
    static BlockingTaskExecutor of() { ... }
    static BlockingTaskExecutorBuilder builder() { ... }
}
```
and removing this `BlockingTaskExecutors` class. 😉

Reply:
That's correct, @minwoox. Gentle ping, @JunoJunho :wink:

Reply:
Sorry for late update 🚨

Reply:
No worries, @JunoJunho. Thank you so much for sticking around. :bow: